File: solutions/c/space-age/1/space_age.c
// Dedicated to Shree DR.MDD

#include "space_age.h"

float age(planet_t orb, int64_t sec) {
    float base = (float)sec / 31557600.0f;

    switch (orb) {
        case MERCURY:
            return base / 0.2408467f;
        case VENUS:
            return base / 0.61519726f;
        case EARTH:
            return base / 1.0f;
        case MARS:
            return base / 1.8808158f;
        case JUPITER:
            return base / 11.862615f;
        case SATURN:
            return base / 29.447498f;
        case URANUS:
            return base / 84.016846f;
        case NEPTUNE:
            return base / 164.79132f;
    }
    return -1;
}

File: solutions/r/space-age/1/space-age.R
space_age <- function(seconds, planet) {
  seconds_in_year <- 31557600 #31557600 = 365.25 * 24 * 60 * 60
  
  orbital_seconds <- switch(tolower(planet),
                   mercury = 0.2408467,
                   venus = 0.61519726,
                   earth = 1,
                   mars = 1.8808158,
                   jupiter = 11.862615,
                   saturn = 29.447498,
                   uranus = 84.016846,
                   neptune = 164.79132)
  
  round(seconds / (orbital_seconds * seconds_in_year), 2)
}